### PR TITLE
add pods logs support in kube runtime

### PR DESCRIPTION
**Pull request by tornado404**
### PR types
New features

### PR changes
Others

### Describe
add pods logs support in kube runtime

### CODE REVIEW

**Review by ElsieFan — APPROVED** [review]

File: pkg/job/runtime_v2/client/kube_runtime_client_test.go [hunk 1]
```diff
@@ -19,6 +19,9 @@ package client
 import (
 	"context"
 	"encoding/json"
+	"github.com/PaddlePaddle/PaddleFlow/pkg/common/utils"
```
Comment (D0m021ng) [inline pkg/job/runtime_v2/client/kube_runtime_client_test.go]: format

File: pkg/job/runtime_v2/kubernetes_runtime.go [hunk 2]
```diff
@@ -250,33 +252,26 @@ func (kr *KubeRuntime) Queue(fwVersion pfschema.FrameworkVersion) framework.Queu
...
-		}
-		queueController := controller.NewQueueSync()
-		err = queueController.Initialize(kr.kubeClient)
-		if err != nil {
-			log.Errorf("init queue controller on %s failed, err: %v", kr.String(), err)
-			return
-		}
-		go jobController.Run(stopCh)
-		go queueController.Run(stopCh)
+	jobController := controller.NewJobSync()
```
Comment (D0m021ng) [inline pkg/job/runtime_v2/kubernetes_runtime.go]: Rebase error/

File: pkg/job/runtime_v2/factory.go [hunk 3]
```diff
@@ -45,6 +45,8 @@ type RuntimeService interface {
 	DeleteJob(job *api.PFJob) error
 	// GetJobLog get log for job
 	GetJobLog(jobLogRequest schema.JobLogRequest) (schema.JobLogInfo, error)
+	// GetMixedLog
```
Comment (D0m021ng) [inline pkg/job/runtime_v2/factory.go]: Merge `GetJobLog` and `GetMixedLog`

File: pkg/common/schema/log.go [hunk 4]
```diff
@@ -28,8 +28,11 @@ type JobLogInfo struct {
 }
 
 type TaskLogInfo struct {
-	TaskID string  `json:"taskID"`
-	Info   LogInfo `json:"logInfo"`
+	// container name
+	TaskID  string  `json:"taskID"`
+	PodName string  `json:"podName"`
```
Comment (D0m021ng) [inline pkg/common/schema/log.go]: `TaskID` is equal to `PodUID` on kubernetes cluster.

File: pkg/job/runtime_v2/kubernetes_runtime.go [hunk 5]
```diff
@@ -508,6 +503,148 @@ func (kr *KubeRuntime) GetJobLog(jobLogRequest pfschema.JobLogRequest) (pfschema
...
+		log.Errorf("list events for resource %s/%s failed, err: %v", namespace, name, err)
+		return response, err
+	}
+	return events.Items, nil
+}
+
+// GetPod get pod by namespace and name
+func (kr *KubeRuntime) GetPod(namespace, name string) (corev1.Pod, error) {
+	log.Debugf("get kubernetes pod: %s/%s", namespace, name)
+	pod, err := kr.clientset().CoreV1().Pods(namespace).Get(context.TODO(), name, metav1.GetOptions{})
```
Comment (D0m021ng) [inline pkg/job/runtime_v2/kubernetes_runtime.go]: It might be better to use `kr.clientset().CoreV1(),Pods(string).Get(x, x, x)`

Comment (tornado404) [reply pkg/job/runtime_v2/kubernetes_runtime.go]: ok

File: pkg/common/schema/log.go (hunk 4 above)
Comment (tornado404) [reply pkg/common/schema/log.go]: done

File: pkg/job/runtime_v2/factory.go (hunk 3 above)
Comment (tornado404) [reply pkg/job/runtime_v2/factory.go]: done

File: pkg/job/runtime_v2/kubernetes_runtime.go (hunk 2 above)
Comment (tornado404) [reply pkg/job/runtime_v2/kubernetes_runtime.go]: fixed

File: pkg/job/runtime_v2/client/kube_runtime_client_test.go (hunk 1 above)
Comment (tornado404) [reply pkg/job/runtime_v2/client/kube_runtime_client_test.go]: done

**Review by D0m021ng — APPROVED** [review]

**Pull request merged**